libosmscout/src/osmscout/util/utf8helper.cpp

```diff
@@ -0,0 +1,276 @@
+#include <osmscout/util/utf8helper.h>
+#include <osmscout/util/utf8helper_charmap.h>
+
+#include <string>
+#include <cstring>
+
+namespace utf8helper
+{
+
+static std::string parse(const std::string& text, codepoint (*func)(const character*, char context)){
```

Comment:
It would be nice to have some description for what the code is doing. Also some external link to a more in deep documentation would be helpful.

What about some tests, especially for invalid input to make sure that corner cases are handled correctly?